apple/t2/default.nix

```
@@ -11,45 +11,96 @@ let
   overrideAudioFiles = package: pluginsPath:
     package.overrideAttrs (new: old: {
       preConfigurePhases = old.preConfigurePhases or [ ] ++ [ "postPatchPhase" ];
-
       postPatchPhase = ''
         cp -r ${audioFiles}/files/{profile-sets,paths} ${pluginsPath}/alsa/mixer/
       '';
     });
+
+  pipewirePackage = overrideAudioFiles pkgs.pipewire "spa/plugins/";
+
+  apple-set-os-loader-installer = pkgs.stdenv.mkDerivation rec {
+    name = "apple-set-os-loader-installer-1.0";
+    src = pkgs.fetchFromGitHub {
+      owner = "Redecorating";
+      repo = "apple_set_os-loader";
+      rev = "r33.9856dc4";
+      sha256 = "hvwqfoF989PfDRrwU0BMi69nFjPeOmSaD6vR6jIRK2Y=";
+    };
+    buildInputs = [ pkgs.gnu-efi ];
+    buildPhase = ''
+      substituteInPlace Makefile --replace "/usr" '$(GNU_EFI)'
+      export GNU_EFI=${pkgs.gnu-efi}
+      make
+    '';
+    installPhase = ''
+      install -D bootx64_silent.efi $out/bootx64.efi
+    '';
+  };
+
+  t2Cfg = config.appleT2Config;
+
 in
 {
-  # For keyboard and touchbar
-  boot.kernelPackages = with pkgs; recurseIntoAttrs (linuxPackagesFor (callPackage ./pkgs/linux-t2.nix { }));
-  boot.initrd.kernelModules = [ "apple-bce" ];
+  options = {
+    appleT2Config.enableAppleSetOsLoader = lib.mkOption {
```

Let's stick to nixpkgs naming conventions (also they are not documented anywhere)

```suggestion
    hardware.apple-t2.enableAppleSetOsLoader = lib.mkOption {
```